src/nghttp.cc

```diff
@@ -414,6 +482,14 @@ int submit_request(HttpClient *client, const Headers &headers, Request *req) {
 
   req->req_nva = std::move(build_headers);
 
+  if (expect_continue) {
+    auto timer = std::make_shared<ContinueTimer>(client->loop, req);
+    timer->start();
+
+    req->continue_timer = timer;
+    client->continue_timers.push_back(timer);
```

Comment:
I tried that first, but IIRC `Request` outlasts the connection itself, which means that the `ev_loop` hangs until every timeout fires.

That said... I'm still not incredibly happy with the lifetime of the `ContinueTimer`. There are two more cases where the current implementation falls short: if the server immediately sends a final response (instead of a 100 Continue), or if the stream itself is reset (say, due to a server error), the timer isn't stopped correctly.

What I'd really like is an object that is tied to the lifetime of the response on the stream, but I didn't see a good candidate. I think I will need to add explicit calls to `timer->stop()` in those two cases.